runtime/lua/vim/ui.lua

```diff
@@ -180,10 +180,9 @@ function M._get_url()
     end
   end
 
-  local old_isfname = vim.o.isfname
-  vim.cmd [[set isfname+=@-@]]
-  local url = vim.fn.expand('<cfile>')
-  vim.o.isfname = old_isfname
+  local url = vim._with({ o = { isfname = vim.o.isfname .. ',@-@' } }, function()
```

Comment:
I'd recommend using more targeted scopes for options. As `isfname` is global option, I'd use `go` instead of `o`.
It seems to make code more explicit while avoiding [extra step in implementation](https://github.com/neovim/neovim/blob/fc9b70826ec88ca2e6c0624c522b872e87aa7ac1/runtime/lua/vim/shared.lua#L1183) that decided which scope should be used.

Reply:
I don't think that's more correct. If `isfname` becomes global-local in the future, won't `vim.go` be incorrect (since we're changing the default/global value rather than the value of the buffer)?

Reply:
*If* 'isfname' scopes are changed, *then* using `bo` or `wo` here will be better. Right now this option is sensible to be used with `go`, as its scope is global. Using `o` as scope here will only lead to more work done during this function execution with the same result as `go`.

That said, It is not particularly strong opinion per se.

Reply:
OK, fixed.